src/test/rules/no-constructor-attributes_test.ts

```diff
@@ -55,6 +55,58 @@ ruleTester.run('no-constructor-attributes', rule, {
       code: `class NonElement {
       constructor() { this.getAttribute('x'); this.appendChild(c); }
     }`
+    },
+    {
+      code: `class Foo extends HTMLElement {
+        constructor() {
+          this.shadowRoot.hasAttribute('foo');
```

Comment:
these methods won't actually exist on a shadow root afaik. the test should probably be that a shadowroot child can be mutated:

```ts
this.shadowRoot.children[0].getAttribute('foo');
```

or something along those lines. 

the rule will still check such an impossible case, but thats ok i think to keep it simple

Reply:
funnily enough just saw they were already here! previous-me made a mistake, good if we can fix that in this branch 😂

Reply:
Yeah, I just copy-pasted the old tests :smile: I'll fix that during the weekend.